### DISCUSSION TITLE
Why can't I use XML to display text?

### DISCUSSION THREAD

**Discussion by cnc-elon** [dotnet/maui#17312] [Q&A] [answered]
The content of the XML file is as follows.
<?xml version="1.0" encoding="utf-8"?>
<LinearLayout
        xmlns:android="http://schemas.android.com/apk/res/android"
		xmlns:tools="http://schemas.android.com/tools"
	    xmlns:maui="clr-namespace:Microsoft.Maui.Controls;assembly=Microsoft.Maui.Controls"
        android:id="@+id/linear_layout"
        android:layout_width="match_parent"
        android:layout_height="match_parent"
        android:background="@android:color/holo_blue_light"
        >

    <TextView
        android:id="@+id/textView1"
        android:layout_width="match_parent"
        android:layout_height="match_parent"
        android:text="Hello,World"
        android:textSize="50sp"
        android:textColor="@android:color/holo_red_light"
        android:cursorVisible="true"/>

      
</LinearLayout>
Text can be displayed on the designer, but not on the emulator.

**Comment by cnc-elon** [comment]
![スクリーンショット 2023-09-12 113127](https://github.com/dotnet/maui/assets/103879360/55e03980-0b11-4dfc-8239-1e0a1c9c3796)

**Comment by cnc-elon** [comment]
public class MainActivity : MauiAppCompatActivity
  {
      public static MainActivity Instance;
}
public class NoticeContentCardViewRenderer : ViewRenderer
{
protected override void OnElementChanged(ElementChangedEventArgs<View> e)
{ 
var activity = (Activity)MainActivity.Instance;
View = activity.LayoutInflater.Inflate(Resource.Layout.TestLayout, this, false);
SetNativeControl(View);
}
}

**Reply by bradencohen** [reply, marked as answer]
Did you register the renderer in your `MauiProgram` file? Also, you should consider using a handler if this is a new control, renderers are considered legacy.

**Reply by cnc-elon** [reply]
Thank you! I tried using handler and the text was displayed

**Reply by headintheclouds21** [reply]
Hi @Vito258 I'm trying to do the same thing that you did. Could you please share the code that worked for you? I know I need to implement my own handler, but I'm still confused what it should look like. It would be greatly appreciated.
